front: drop TimeButtons, TrainDetails and ChartSynchronizer

emersion opened the pull request:
These are unused at the moment, because the whole thing is hidden. The code is bit-rotting and holding off refactoring.

Let's just nuke all of that. Maybe this feature will be re-introduced at some point, but we'll need to rewrite big chunks of it anyways and we can easily copy back useful bits from the Git history.

emersion committed:
front: drop TimeButtons, TrainDetails and ChartSynchronizer

These are unused at the moment, because the whole thing is hidden.
The code is bit-rotting and holding off refactoring.

Let's just nuke all of that. Maybe this feature will be re-introduced
at some point, but we'll need to rewrite big chunks of it anyways
and we can easily copy back useful bits from the Git history.

Signed-off-by: Simon Ser <contact@emersion.fr>

Wadjetz reviewed (approved):
lgtm

clarani commented on front/src/modules/simulationResult/components/TimeButtons.tsx:
```diff
@@ -1,171 +0,0 @@
-import InputSNCF from 'common/BootstrapSNCF/InputSNCF';
-import { updateIsPlaying } from 'reducers/simulationResults';
-import { useAppDispatch } from 'store';
-import { datetime2time, sec2datetime, time2datetime } from 'utils/timeManipulation';
```
You can remove time2datetime in `utils/timeManipulation`

clarani reviewed (commented):
You can also:
- drop all the utils functions interpolateOnTime, interpolateOnPosition
- drop getSimulationHoverPositions
- clean SimulationWarpedMap

emersion committed:
front: drop time2datetime()

This is now unused.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion committed:
front: drop getSimulationHoverPositions()

This is now unused.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion committed:
front: drop ChartHelpers

These are now unused.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion committed:
front: drop outdated train hover logic from SimulationWarpedMap

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion commented:
All done!

clarani commented on front/src/common/Map/WarpedMap/SimulationWarpedMap.tsx:
You forgot 2 commented lines at the bottom of this file

clarani commented on front/src/modules/simulationResult/components/ChartHelpers/ChartHelpers.ts:
You can also drop sec2d3datetime and getAxis

clarani commented on front/src/modules/simulationResult/components/SimulationResultsMap/helpers.ts:
You can drop this file

emersion committed:
front: drop WarpedMapProps.trainsPositions

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion committed:
front: drop TrainHoverPosition

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion commented:
Removed! I also ended up nuking `TrainHoverPosition`.

clarani reviewed (commented):
You can drop:
- the file `src/reducers/simulationResults/types.ts` and most of consts in `src/modules/simulationResult/conts.ts`
- the file trainExample.ts
- the file simulationTrain.ts

(You can also drop TrainOnMap)

Synar commented:
If you end up dropping TrainOnMap, please also drop the train layer and layer group in front/src/config/layerOrder.ts ^^

emersion committed:
front: drop unused JSON assets

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion committed:
front: drop unused types and consts from simulationResults

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion committed:
front: drop TrainOnMap

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion commented:
- `src/reducers/simulationResults/types.ts`: had to keep some of the types (used by warped map)
- `src/modules/simulationResult/conts.ts`: ditto (used by the space time chart)
- `trainExample.ts`, `simulationTrain.ts`: dropped
- `TrainOnMap` (incl layer): dropped

clarani reviewed (approved):
You can drop getCurrentBearing too

emersion committed:
front: drop getCurrentBearing()

This is now unused.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion commented:
> You can drop getCurrentBearing too

Done

clarani reviewed (approved):
LGTM :white_check_mark:
